File: python/2956.py
from typing import List

class Solution:
    def findIntersectionValues(self, nums1: List[int], nums2: List[int]) -> List[int]:
        s1 = set(nums1)
        s2 = set(nums2)
        a1 = sum(x in s2 for x in nums1)
        a2 = sum(x in s1 for x in nums2)
        return [a1, a2]

File: go/2956.go
package main

func findIntersectionValues(nums1 []int, nums2 []int) []int {
	s1 := map[int]int{}
	s2 := map[int]int{}
	for _, num := range nums1 {
		s1[num] = 1
	}
	for _, num := range nums2 {
		s2[num] = 1
	}
	ans := [2]int{}
	for _, num := range nums1 {
		ans[0] += s2[num]
	}
	for _, num := range nums2 {
		ans[1] += s1[num]
	}
	return ans[:]
}
